### PR TITLE
Employee register done

### PR DESCRIPTION
**Pull request by MrCano369x**
Co-authored-by: alanRazoMen <alanRazoMen@users.noreply.github.com>
Co-authored-by: AlejandroCuevasZepeda <AlejandroCuevasZepeda@users.noreply.github.com>
Co-authored-by: Alex Silva <alexs@users.noreply.github.com>

# Description
we create a scaffold with the attendances model,

In controller:
first we check if the employee exists
we verify if the attendance is a check in or a check out

  ## Screenshots
gif

### CODE REVIEW

File: app/controllers/attendances_controller.rb [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+        nuevo = {"check_in" => DateTime.now()}.merge(attendance_params)
+        @attendance = Attendance.new(nuevo)
+
+        if @attendance.save
+          format.html { redirect_to attendance_url(@attendance), notice: "Se registró tu Entrada" }
+        else
+          format.html { render :new, status: :unprocessable_entity }
+        end
+
+      end
```
Comment (Frcomparan) [inline app/controllers/attendances_controller.rb:54]: In this block, a record is chosen in Attendance that matches with `:employee_private_number`, once the `check_in` and `check_out` records are saved show the message " Ya te registraste por hoy". Doesn't this create a problem for you where you can save a maximum of one record per employee?

Comment (alexs) [reply app/controllers/attendances_controller.rb:54]: You are right, but right now it's a first mvp we will refactor and improve in the next iterations spring. 😀

Comment (alexs) [reply app/controllers/attendances_controller.rb:54]: Yep! You are right, but for now it's just a first mvp we will refactor and improve in the next iterations spring. 😀

File: app/helpers/attendances_helper.rb [hunk 2]
```diff
@@ -0,0 +1,2 @@
+module AttendancesHelper
+end
```
Comment (Frcomparan) [inline app/helpers/attendances_helper.rb:2]: If you don't use this file, you should delete it

**Review by Frcomparan — DISMISSED** [review]
LGTM!

File: app/controllers/attendances_controller.rb [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+        puts "no hay"
+        nuevo = {"check_in" => DateTime.now()}.merge(attendance_params)
+        @attendance = Attendance.new(nuevo)
+
+        if @attendance.save
+          format.html { redirect_to attendance_url(@attendance), notice: "Se registró tu Entrada" }
+        else
+          format.html { render :new, status: :unprocessable_entity }
+        end
+
```
Comment (dannharo) [inline app/controllers/attendances_controller.rb:53]: why are you using `puts` in the controller?
you could use flash messages o loggers for the messages, also remove the comments

File: app/views/attendances/index.html.erb [hunk 4]
```diff
@@ -0,0 +1,21 @@
+<p style="color: green"><%= notice %></p>
```
Comment (dannharo) [inline app/views/attendances/index.html.erb:1]: avoid using inline styles.

File: app/views/attendances/show.html.erb [hunk 5]
```diff
@@ -0,0 +1,5 @@
+<p style="color: green"><%= notice %></p>
```
Comment (dannharo) [inline app/views/attendances/show.html.erb:1]: also here, you could create a class with this property and use it

File: app/views/layouts/_menu.html.erb [hunk 6]
```diff
@@ -13,7 +13,10 @@
           <%= link_to "Employees", employees_path, class: "nav-link #{controller_name == "employees" ? 'active' : '' }" %>
         </li>
         <li class="nav-item">
-          <%= link_to "Company_branches", company_branches_path, class: "nav-link #{controller_name == "company_branches" ? 'active' : '' }" %>
+          <%= link_to "Company branches", company_branches_path, class: "nav-link #{controller_name == "company_branches" ? 'active' : '' }" %>
```
Comment (dannharo) [inline app/views/layouts/_menu.html.erb:16]: I recommend you use single quotes when you use strings on ruby.

**Review by dannharo — COMMENTED** [review]
good job, I recommend you to only use one language, I see Spanish and english mixed.

also you should read about the rails associations in `Active Records`
https://guides.rubyonrails.org/association_basics.html
you could create the associations `attendances-employees`, IDK why you are using the `employee_private_number` to the relation 🤔

File: app/controllers/attendances_controller.rb (hunk 3 above)
Comment (MrCano369x) [reply app/controllers/attendances_controller.rb]: we were testing and we forgot to delete it but we have refactored this

File: app/views/layouts/_menu.html.erb (hunk 6 above)
Comment (MrCano369x) [reply app/views/layouts/_menu.html.erb]: changed

File: app/views/attendances/show.html.erb (hunk 5 above)
Comment (MrCano369x) [reply app/views/attendances/show.html.erb]: changed

File: app/views/attendances/index.html.erb (hunk 4 above)
Comment (MrCano369x) [reply app/views/attendances/index.html.erb]: changed

**Review by Frcomparan — APPROVED** [review]

**Review by Javi1499 — APPROVED** [review]

**Review by EliasUR — APPROVED** [review]

File: app/controllers/attendances_controller.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (MrCano369x) [reply app/controllers/attendances_controller.rb]: this file was fixed yesterday :(

File: app/controllers/attendances_controller.rb [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+  # POST /attendances or /attendances.json
+  def create
+    respond_to do |format|
+      employee = Employee.find_by(private_number: attendance_params[:employee_private_number])
+      if employee == nil
+        puts "El empleado no existe"
+        # flash.notice = "pinga"
+        return
+        # return format.html { redirect_to attendance_url(registro), notice: "No existes xd" }
```
[the remark this answers is not in the document]
Comment (MrCano369x) [reply app/controllers/attendances_controller.rb]: file outdated >:u

**Pull request merged**